src/main/java/com/github/kaklakariada/fritzbox/http/HttpTemplate.java

```diff
@@ -94,22 +94,10 @@ private <T> T parse(final Response response, Class<T> resultType) {
         if (!response.isSuccessful()) {
             throw new FritzBoxException("Request failed: " + response);
         }
-        final String body = getBodyAsString(response);
-        if (body.startsWith("HTTP/1.0 500 Internal Server Error")) {
-            throw new FritzBoxException("Request failed: " + body);
-        }
-        if (LOG.isTraceEnabled()) {
-            LOG.trace("Got response {} with body\n'{}'", response, body.trim());
-        }
-        return deserializer.parse(body.trim(), resultType);
-    }
-
-    private String getBodyAsString(final Response response) {
-        try {
-            return response.body().string();
-        } catch (final IOException e) {
-            throw new HttpException("Error getting body from response " + response, e);
+        if (response.code() == 500) {
```

Comment:
```suggestion
        if (response.code() != 200) {
```
:question:

Reply:
@kaklakariada In most cases this should work I think. There are some other HTTP response codes that also indicates successful responses. I think it depends on a. what the Fritzbox can send and b. what codes can be send if the fritzbox is behind a proxy.
Changing the response code handling is another aspect so from my point of view this should be changed in a seperate PR / branch. Checking the response code value instead of looking for a string in the body was made out of necessity because now there is a stream and not a string to disposal. Apart from that it is even a more solid solution because the string "HTTP/1.0 500 Internal Server Error" could be changed in future, while response code remains.

Reply:
That's OK for me.Thank you!

Reply:
You are welcome!